src/java.base/windows/native/libjava/WinNTFileSystem_md.c

```diff
@@ -469,13 +469,16 @@ Java_java_io_WinNTFileSystem_setPermission0(JNIEnv *env, jobject this,
     jboolean rv = JNI_FALSE;
     WCHAR *pathbuf;
     DWORD a;
-    if (access == java_io_FileSystem_ACCESS_READ ||
-        access == java_io_FileSystem_ACCESS_EXECUTE) {
-        return enable;
-    }
+
     pathbuf = fileToNTPath(env, file, ids.path);
     if (pathbuf == NULL)
         return JNI_FALSE;
+
+    if (access == java_io_FileSystem_ACCESS_READ ||
+        access == java_io_FileSystem_ACCESS_EXECUTE) {
+        return _waccess(pathbuf, 0) == 0 ? enable : JNI_FALSE;
```

Comment:
Here `enable` is returned for backward compatibility, but per the specification it seems that `JNI_TRUE` should be returned instead.

Reply:
I don't think this is right, at least it doesn't work with ACLs and file system security so it can't test if the file is executable. Also I have a concern about mixing win32 and C runtime functions here.  The main issue with these setXXX methods is that don't map to DOS file attributes or ACL based security so they will need to fail for some cases.

Reply:
If this code does not interact with ACL-based security, then it's not clear to me that there's anything to be done here.

Reply:
> If this code does not interact with ACL-based security, then it's not clear to me that there's anything to be done here.

Right, these methods were created with POSIX file permissions in mind. On the surface, the only result it can return is false (to mean fail) but the concern is that it would be an incompatible change so we need to think a bit more about it.

Reply:
Commit 08c5fd936e79fe1d08ca5c7fc9cc59a8c71ff555 reverts the impl changes in favor of adjusting the spec to match behavior. This PR will be tagged and a CSR created once there is consensus on the spec update.

Note that the spec of `File.setWritable` is _not_ changed.